falcon/cluster/cluster.py

```diff
@@ -920,3 +746,76 @@ def _get_cluster_medoid_index(
         if row_avg < min_avg:
             min_i, min_avg = row_i, row_avg
     return cluster_mask[min_i]
+
+
+def compute_condensed_distance_matrix(
+    spec_tuples: List[similarity.SpectrumTuple],
+    fragment_mz_tol: float,
+    allow_shift: bool,
+) -> np.ndarray:
+    """
+    Compute the condensed pairwise distance matrix for the given spectra.
+
+    Parameters
+    ----------
+    spec_tuples : List[similarity.SpectrumTuple]
+        The spectra to compute the pairwise distance matrix for.
+    fragment_mz_tolerance : float
+        The fragment m/z tolerance.
+    allow_shift : bool
```

Comment:
**todo:** This can be removed. During clustering we're only considering near-identical spectra from the same molecule, no shifted peaks.